http-filter-example/http_filter.cc

```diff
@@ -26,9 +29,45 @@ const std::string HttpSampleDecoderFilter::headerValue() const {
   return config_->val();
 }
 
+int HttpSampleDecoderFilter::headerExtra() const {
+  return config_->extra();
+}
+
 FilterHeadersStatus HttpSampleDecoderFilter::decodeHeaders(RequestHeaderMap& headers, bool) {
-  // add a header
-  headers.addCopy(headerKey(), headerValue());
+
+  // parse config: key is header operation, value is a vector of arguments for the operation
+  std::unordered_map<std::string, std::vector<std::string>> header_ops;
+  std::string header_config = headerValue();
+
+  // Find the position of the first space character
+  size_t spacePos = header_config.find(' ');
+  std::string operation = header_config.substr(0, spacePos);
```

Comment:
Nitpick: If you use `auto` here instead of `std::string`, `substr` in C++14 returns a `string_view`, which will avoid an unnecessary copy

Reply:
I think these variables can also be `const`